### PR TITLE
Add subscribe docs

### LINKED ISSUE #4974: Subscribe Documentation

### PR DESCRIPTION
**Pull request by nikoulai**
## Description

closes #4974 

## Checklist for 4.x:

- [ ] I have selected the correct 4.x base branch.
- [ ] Within the description, the feature or issue is discussed in detail or an issue is linked.   
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added any required tests for the changes I made 
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `yarn` successfully
- [ ] I ran `yarn lint` successfully
- [ ] I ran `yarn build:web` successfully
- [ ] I ran `yarn test:unit` successfully
- [ ] I ran `yarn test:integration` successfully
- [ ] I ran `compile:contracts` successfully
- [ ] I have tested my code.
- [ ] I have updated the corresponding `CHANGELOG.md` file in the packages I have edited.

### CODE REVIEW

File: packages/web3-eth/src/web3_subscriptions.ts [hunk 1]
```diff
@@ -58,6 +67,16 @@ export class LogsSubscription extends Web3Subscription<
 	}
 }
 
+/**
+ * ##subscribe(“pendingTransactions”)
```
Comment (avkos) [inline packages/web3-eth/src/web3_subscriptions.ts]: Headers do not work here
image

**Review by avkos — CHANGES_REQUESTED** [review]

File: packages/web3-eth/src/web3_subscriptions.ts (hunk 1 above)
Comment (nikoulai) [reply packages/web3-eth/src/web3_subscriptions.ts]: fixed

File: packages/web3-eth/src/index.ts [hunk 2]
```diff
@@ -41,6 +41,9 @@ along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
  * const web3 = new Web3(Web3.givenProvider || 'ws://some.local-or-remote.node:8546');
  * web3.eth.getBalance('0x407d73d8a49eeb85d32cf465507dd71d507100c1').then(console.log);
  * > 1000000000000n
+ *```
+ *
+ * With `web3-eth` you can also subscribe (if supported by provider) to events in the Ethereum Blockchain, using the `subscribe` function. See more at the {@link subscribe} function.
```
Comment (luu-alex) [inline packages/web3-eth/src/index.ts]: Screenshot 2023-03-17 at 12 15 58 PM
the link doesnt seem to be working, would u know why?

File: packages/web3-eth/src/web3_eth.ts [hunk 3]
```diff
@@ -1519,8 +1519,21 @@ export class Web3Eth extends Web3Context<Web3EthExecutionAPI, RegisteredSubscrip
 	 *
 	 * @param name The subscription you want to subscribe to.
 	 * @param args Optional additional parameters, depending on the subscription type.
-	 * @returns One of {@link RegisteredSubscription}
+	 * @returns A subscription object of type {@link RegisteredSubscription}. In the object there are present:
```
Comment (luu-alex) [inline packages/web3-eth/src/web3_eth.ts]:
```suggestion
	 * @returns A subscription object of type {@link RegisteredSubscription}. The object contains:
```

File: packages/web3-eth/src/web3_subscriptions.ts [hunk 4]
```diff
@@ -77,6 +96,35 @@ export class NewPendingTransactionsSubscription extends Web3Subscription<
 	}
 }
 
+/**
+ * ## subscribe(“newHeads”)
```
Comment (luu-alex) [inline packages/web3-eth/src/web3_subscriptions.ts]: From my understanding `subscribe("newHeads")` and `subscribe("newBlockHeaders")` is the same, can we mention that? As well show the example of `web3.eth.subscribe("newHeads")`

**Comment by luu-alex** [comment]
Can we get docs for [syncingsubscription](https://github.com/web3/web3.js/blob/6e0c8638f26f76f4ab47999c38cfb9e139ea2b05/packages/web3-eth/src/web3_subscriptions.ts#L147)

File: packages/web3-eth/src/index.ts (hunk 2 above)
Comment (nikoulai) [reply packages/web3-eth/src/index.ts]: I didn't use a link, just bolded the text. Would a link be useful (since we are inside the same package)?

Comment (luu-alex) [reply packages/web3-eth/src/index.ts]: `See more at the {@link subscribe} function.` {@link subscribe} i think when you use link here, subscribe should redirect to the subscribe function

File: packages/web3-eth/src/web3_subscriptions.ts [hunk 5]
```diff
@@ -96,6 +144,11 @@ export class NewHeadsSubscription extends Web3Subscription<
 	}
 }
 
+/**
+ * ## subscribe(“syncing”)
+ *
+ * Subscribe to syncing events. This will return an object when the node is syncing and when it’s finished syncing will return `false`
+ */
```
Comment (avkos) [inline packages/web3-eth/src/web3_subscriptions.ts:165]: it has changed event = true/false
and data event
example here packages/web3/test/integration/sync.test.ts

**Review by avkos — CHANGES_REQUESTED** [review]
please add example for syncing event

**Review by avkos — APPROVED** [review]

**Review by luu-alex — APPROVED** [review]

**Pull request merged**